### PR TITLE
Philipp Schöneberg Abgabe 2

### PR DESCRIPTION
**Pull request by PhilippSchoeneberg**

### CODE REVIEW

File: Exercises/Philipp Schöneberg/exercises_02_solved.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            case 2:
+                lst[lst[i*4+3]] = lst[lst[i*4+1]]*lst[lst[i*4+2]]
+            case 99:
+                break
+            case _:
+                print("The given opcode is unvalid.")
+    return lst[0]
+
+print(list_manipulator(commands))
+
```
Comment (HannahGedler) [inline Exercises/Philipp Schöneberg/exercises_02_solved.py:46]: I'm really impressed by the (short) length of your code

File: Exercises/Philipp Schöneberg/exercises_02_solved.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# All inputs will be of type string.
+# the function should return two lists:
+#   The first list should contain all arguments which can be interpreted
+#   as a number.
+#   The second list should contain all strings which contain just one character.
+# Think of some good inputs to test this functionality, write down at least three
+# examples and verify that the output for these examples is correct.
+
+def string_sorter (*args):
+    lst_0 = []
```
Comment (HannahGedler) [inline Exercises/Philipp Schöneberg/exercises_02_solved.py:58]: maybe you could name these two lists more specific, at first I was confused which list should contain the numbers and which the one-character-strings

File: Exercises/Philipp Schöneberg/exercises_02_solved.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+print(lst_numbers)
+print(lst_mono_character)
+
+lst_numbers, lst_mono_character = string_sorter("1", "2.0", "22", ".", "x", "HalloWelt", "Hello World")
+print(lst_numbers)
+print(lst_mono_character)
+
+lst_numbers, lst_mono_character = string_sorter("1", "2.0", "22", ".", "x", "HalloWelt", "Hello World", " ", "")
+print(lst_numbers)
+print(lst_mono_character)
```
Comment (HannahGedler) [inline Exercises/Philipp Schöneberg/exercises_02_solved.py:80]: maybe you could insert the print statements in the function and then call lst_0 or lst_1 so you don't have to write it three times

Comment (PhilippSchoeneberg) [reply Exercises/Philipp Schöneberg/exercises_02_solved.py]: Thank you for this good idea.

File: Exercises/Philipp Schöneberg/exercises_02_solved.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    for i, elem in enumerate(lst[::4]):
+        match lst[i*4]:
+            case 1:
+                lst[lst[i*4+3]] = lst[lst[i*4+1]]+lst[lst[i*4+2]]
+            case 2:
+                lst[lst[i*4+3]] = lst[lst[i*4+1]]*lst[lst[i*4+2]]
+            case 99:
+                break
+            case _:
+                print("The given opcode is unvalid.")
```
Comment (Saiki77) [inline Exercises/Philipp Schöneberg/exercises_02_solved.py:42]: Maybe raise an error like you did for the second task.

File: Exercises/Philipp Schöneberg/exercises_02_solved.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            pass
+        if(len(i) == 1):
+            lst_mono_character.append(i)
+    print(lst_numbers)
+    print(lst_mono_character)
+    return lst_numbers, lst_mono_character
+
+string_sorter("1", "2.0", "22")
+string_sorter("1", "2.0", "22", ".", "x", "HalloWelt", "Hello World")
+string_sorter("1", "2.0", "22", ".", "x", "HalloWelt", "Hello World", " ", "")
```
Comment (Saiki77) [inline Exercises/Philipp Schöneberg/exercises_02_solved.py:75]: Very good to think of edge cases such as "" and " "

**Review by Saiki77 — APPROVED** [review]

File: Exercises/Philipp Schöneberg/exercises_02_solved.py (hunk 4 above)
Comment (PhilippSchoeneberg) [reply Exercises/Philipp Schöneberg/exercises_02_solved.py]: Good idea I just did. Thx!

**Pull request merged**